src/repository.h

```diff
@@ -160,7 +160,9 @@ struct git_repository {
 
 GIT_INLINE(git_attr_cache *) git_repository_attr_cache(git_repository *repo)
 {
-	return repo->attrcache;
+	git_attr_cache *cache;
+	__atomic_load(&repo->attrcache, &cache, __ATOMIC_ACQUIRE);
```

Comment:
By the way, there are synchronization bugs when attrcache is lazily initialized. I added a [workaround](https://github.com/romkatv/gitstatus/blob/d3ce9ef43951fcf6eb04cbb005a06f359fb78610/src/repo.cc#L109) in gitstatusd instead of fixing it in libgit2.